Unit tests for Cordova platform

mdineen opened the pull request:
Please review unit tests.

Updated the title of this pull request to better reflect the work that was done in it

Commit:
Gord updated some test code to safely mock _db in all cases

Weekend workaround for the jake test errors found in platform.js, defaults _current if undefined

undid sinful default

cordova geolocation tests add

added camera.js to unit tests - cant invoke due to UI dependency

first draft of geo and contacts unit tests

changed machines and had to set geo.timeout false beforeEach()

switching back to the desktop

provided local mock for contacts

updated cordova version to 2.0.0, added unit tests for capture.js, console.js, device.js

fixed a tab space on contacts.  added network.js

added unit tests for cordova/network

added unit tests for cordova noticiation

fixed timing issue in accelerometer tests

added FileError.js so it could be referenced in ripple

first commit of the changes to file.js and its tests

completed unit tests for file.js, modified file.js to use window.FileReader, rather than just FileReader in a couple of spots.

added tests for bridge to camera.js

completed round 1 of code review changes

completed round 2 of code review changes

completed round 3 of code review changes

removed FileError.js

removed a commented line

put the tests back.

restored no-args test for vibrate

getting started on the geo-gpx stuff

lint fix

dansilivestru commented on lib/ripple/platform/cordova/2.0.0/FileError.js:
```diff
@@ -0,0 +1,25 @@
+FileError.TYPE_MISMATCH_ERR = 11;
+FileError.PATH_EXISTS_ERR = 12;
+
+module.exports = FileError;
```
This shouldn't really be here. You should fake it in your test. The reason is that FileError will be there in cordova.js

dansilivestru commented on test/unit/cordova/camera.js:
```diff
@@ -0,0 +1,70 @@
+        });
+
+        //blows up
+        xit("can be called with no args", function () {
```
Let's leave it x'd out and open a techdebt to determine what the behaviour actually is on the device. Once we know that, we can emulate it properly.

dansilivestru commented on test/unit/cordova/capture.js:
```diff
@@ -0,0 +1,70 @@
+        });
+
+        //this actually blows up quite handsomely on my machine.  Should throw I suspect.
+        xit("can be called with no args", function () {
```
Let's leave it x'd out and open a techdebt to determine what the behaviour actually is on the device. Once we know that, we can emulate it properly.

dansilivestru commented on test/unit/webworks.tablet/systemEvent.js:
```diff
@@ -226,7 +226,7 @@ describe("webworks.tablet system event", function () {
+            xit("passes baton when battery level changes", function () {
                 var baton = new MockBaton();
                 server.deviceBatteryLevelChange({}, {}, baton);
                 event.trigger("DeviceBatteryLevelChanged", [80], true);
```
let's un-x these?

Commit:
first draft of driving directions for gpx route files capabilities completed

first draft of driving directions for gpx route files capabilities completed

filling in long spaces in the YOURS routing GPX files

Updated manifest.json for chromestore ext for manifest_version: 2

back to v1

enclosed the navigation helpers within a utils object, and switched from "for" to "utils.forEach" for parsing gpx files

enclosed the navigation helpers within a utils object, and switched from "for" to "utils.forEach" for parsing gpx files

readded pesky camera/capture test
